common-custom-user-data-maven-extension/src/main/java/com/gradle/maven/common/user/data/extension/CommonCustomUserDataGradleEnterpriseListener.java

```diff
@@ -1,60 +1,58 @@
+        hint = "common-custom-user-data",
+        description = "Captures common custom user data in Maven build scans"
 )
-public final class CommonCustomUserDataMavenExtension extends AbstractMavenLifecycleParticipant {
+public final class CommonCustomUserDataGradleEnterpriseListener implements GradleEnterpriseListener {
 
-    private final PlexusContainer container;
     private final Logger logger;
 
+    @SuppressWarnings("CdiInjectionPointsInspection")
     @Inject
-    public CommonCustomUserDataMavenExtension(PlexusContainer container, Logger logger) {
-        this.container = container;
+    public CommonCustomUserDataGradleEnterpriseListener(Logger logger) {
         this.logger = logger;
     }
 
     @Override
-    public void afterProjectsRead(MavenSession session) throws MavenExecutionException {
-        logger.debug("Executing extension: " + getClass().getSimpleName());
-
-        GradleEnterpriseApi gradleEnterprise = ApiAccessor.lookupGradleEnterpriseApi(container, getClass());
-        if (gradleEnterprise != null) {
+    public void configure(GradleEnterpriseApi api, @Nullable MavenSession session) {
+        try {
+            logger.debug("Executing extension: " + getClass().getSimpleName());
             CustomGradleEnterpriseConfig customGradleEnterpriseConfig = new CustomGradleEnterpriseConfig();
 
             logger.debug("Configuring Gradle Enterprise");
-            customGradleEnterpriseConfig.configureGradleEnterprise(gradleEnterprise);
+            customGradleEnterpriseConfig.configureGradleEnterprise(api);
             logger.debug("Finished configuring Gradle Enterprise");
 
             logger.debug("Configuring build scan publishing and applying build scan enhancements");
-            BuildScanApi buildScan = gradleEnterprise.getBuildScan();
+            BuildScanApi buildScan = api.getBuildScan();
             customGradleEnterpriseConfig.configureBuildScanPublishing(buildScan);
             new CustomBuildScanEnhancements(buildScan, session).apply();
             logger.debug("Finished configuring build scan publishing and applying build scan enhancements");
 
             logger.debug("Configuring build cache");
-            BuildCacheApi buildCache = gradleEnterprise.getBuildCache();
+            BuildCacheApi buildCache = api.getBuildCache();
             customGradleEnterpriseConfig.configureBuildCache(buildCache);
             logger.debug("Finished configuring build cache");
 
-            GroovyScriptUserData.evaluate(session, gradleEnterprise, logger);
+            GroovyScriptUserData.evaluate(session, api, logger);
             SystemPropertyOverrides.configureBuildCache(buildCache);
-        } else {
-            logger.debug("Could not find GradleEnterpriseApi component in Plexus container");
+        } catch (MavenExecutionException e) {
+            logger.error("Error executing Common Custom User Data Maven extension:");
+            logger.error(e.getMessage(), e);
```

Comment:
Should we maybe handle this at the call site in the GE Maven extension?

Reply:
yes, I think this is a better idea. If the others are ok, I will do this.
Basically `GradleEnterpriseListener#configure` will declare that it can throw a `MavenExecutionException`, and we'll catch it around each listener call on the GE Maven extension side.

Reply:
done